src/java_tools/junitrunner/java/com/google/testing/junit/runner/model/TestSuiteModel.java

```diff
@@ -183,6 +184,18 @@ public void testFailure(Description description, Throwable throwable) {
         test.testFailure(throwable, now());
       }
     }
+    else {
+      final Optional<Description> matchingSuite = testsMap.keySet().stream()
```

Comment:
I think it's worth adding a comment indicating how this scenario occurs and what the motivation for the code is.